internal/common/common.go

```diff
@@ -183,22 +198,42 @@ func fromAmber(provenance *types.ValidatedProvenance) (*ProvenanceIR, error) {
 		return nil, fmt.Errorf("could get builder image digest from *amber.ValidatedProvenance: %v", err)
 	}
 
+	// We collect repo uris from where they appear in the provenance to verify that they point to the same reference repo uri.
+	repoURIs := []string{}
+	repoURIs = append(repoURIs, slsav02.GetConfigSourceURI(*predicate))
+	repoURIs = append(repoURIs, slsav02.GetMaterialsGitURI(*predicate)...)
+
 	provenanceIR := NewProvenanceIR(binarySHA256Digest,
 		WithBuildType(buildType),
 		WithBuildCmd(buildCmd),
-		WithBuilderImageSHA256Digest(builderImageDigest))
+		WithBuilderImageSHA256Digest(builderImageDigest),
+		WithRepoURIs(repoURIs))
 
 	return provenanceIR, nil
 }
 
 // fromSLSAv02 maps data from a validated SLSA v0.2 provenance to ProvenanceIR.
-func fromSLSAv02(provenance *types.ValidatedProvenance) *ProvenanceIR {
+func fromSLSAv02(provenance *types.ValidatedProvenance) (*ProvenanceIR, error) {
 	// A slsa.ValidatedProvenance contains a SHA256 hash of a single subject.
 	binarySHA256Digest := provenance.GetBinarySHA256Digest()
 	buildType := slsav02.GenericSLSABuildType
+
+	predicate, err := slsav02.ParseSLSAv02Predicate(provenance.GetProvenance().Predicate)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse provenance predicate: %v", err)
+	}
+
+	// We collect repo uris from where they appear in the provenance to verify that they point to the same reference repo uri.
+	repoURIs := []string{}
+	repoURIs = append(repoURIs, slsav02.GetConfigSourceURI(*predicate))
+	repoURIs = append(repoURIs, slsav02.GetMaterialsGitURI(*predicate)...)
+
 	provenanceIR := NewProvenanceIR(binarySHA256Digest,
-		WithBuildType(buildType))
-	return provenanceIR
+		WithBuildType(buildType),
+		// TODO(#202): Add WithBuilderImageSHA256Digest.
```

Comment:
I am not sure if there is a `BuilderImageSHA256Digest` for `slsav02.GenericSLSABuildType`, so you can remove the TODO.